.github/workflows/test-release.yml

```diff
@@ -0,0 +1,76 @@
+name: GraalVM Native Image builds
+on: push
+jobs:
+  build-natives:
+    name: Build native images
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, macos-latest, ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: feature/release-workflow
+
+      - uses: graalvm/setup-graalvm@v1
+        with:
+          java-version: '17.0.7'
+          distribution: 'graalvm'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          native-image-job-reports: 'true'
+
+      - name: Build native images with maven plugin
+        shell: bash
+        run: |
+          maven_config="$(cat .mvn/maven.config)"
+          current_version="${maven_config/#*-Drevision=}"
+          current_version="${current_version/ */}"
+          cd cli
+          mvn -B -ntp -Pnative -DskipTests=true package
+          cd target
+          tar -cvf cli-${current_version}-${{ runner.os }}.tar cli*
```

Comment:
IMHO it would be better to solve all such things in maven rather than in workflows (see also my comment that we create a copy of this workflow and would then have such logic redundantly).